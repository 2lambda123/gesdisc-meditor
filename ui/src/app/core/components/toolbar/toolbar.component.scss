.spacer {
    flex: 1 1 auto;
}

.logo {
  height: 80px;
}

.med-toolbar {
  background: none;
  height: 100%;
}

.tb-button {
    margin-left: 10px;
    color: grey;
}

.links_menu {
  width: 100px;
}