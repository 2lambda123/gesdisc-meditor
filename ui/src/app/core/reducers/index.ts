import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
