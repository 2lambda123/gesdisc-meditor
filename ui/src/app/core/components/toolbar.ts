import { Component, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'med-toolbar',
  template: `
    <mat-toolbar class="med-toolbar">
      <img src="assets/logo.png" class="logo"/>
      <span class="spacer"></span>
      <button mat-raised-button color="primary" class="toolbar-btn">
        <mat-icon>feedback</mat-icon>
        Feedback
      </button>
      <button mat-raised-button color="primary" class="toolbar-btn">
        <mat-icon>help</mat-icon>
        Help
      </button>
      <med-login></med-login>
    </mat-toolbar>
  `,
  styles: [`
    .spacer {
      flex: 1 1 auto;
    }

    .logo {
      max-height: 100%;
    }

    .med-toolbar {
    	background: none;
    }

    .toolbar-btn {
    	margin-right: 15px;
    }
  `]
})
export class ToolbarComponent {
  @Output() openMenu = new EventEmitter();
}
