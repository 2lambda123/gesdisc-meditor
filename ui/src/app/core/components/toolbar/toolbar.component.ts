import { Component } from '@angular/core';

@Component({
	selector: 'med-toolbar',
	template: `
		<mat-toolbar class="med-toolbar">
			<img src="assets/logo.png" class="logo" routerLink="/" />
			<span class="spacer"></span>
			<med-login-status></med-login-status>

      <mat-menu #appMenu="matMenu" fxHide.gt-sm>
        <button mat-menu-item routerLink="/">
          <mat-icon>home</mat-icon>
          Home
        </button>
        <a mat-menu-item href="mailto:gsfc-uui-dev-disc@lists.nasa.gov">
          <mat-icon>feedback</mat-icon>
          Feedback
        </a>
        <button mat-menu-item>
          <mat-icon>help</mat-icon>
          Help
        </button>
      </mat-menu>
      <button mat-icon-button [matMenuTriggerFor]="appMenu" fxHide.gt-sm>
        <mat-icon>more_vert</mat-icon>
      </button>
      
      <div fxHide.lt-md>
        <button mat-button class="tb-button" routerLink="/">
          <mat-icon>home</mat-icon>
          Home
        </button>
        <a mat-button class="tb-button" href="mailto:gsfc-uui-dev-disc@lists.nasa.gov">
          <mat-icon>feedback</mat-icon>
          Feedback
        </a>
        <button mat-button class="tb-button">
          <mat-icon>help</mat-icon>
          Help
        </button>
      </div>
		</mat-toolbar>
	`,
	styleUrls: ['toolbar.component.scss']
})
export class ToolbarComponent {
}
