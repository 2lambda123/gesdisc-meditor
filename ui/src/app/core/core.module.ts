import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
