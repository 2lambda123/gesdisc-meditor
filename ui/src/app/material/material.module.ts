import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
