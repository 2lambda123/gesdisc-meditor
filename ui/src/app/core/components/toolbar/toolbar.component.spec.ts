import { ComponentFixture, TestBed  } from '@angular/core/testing';

import { ToolbarComponent } from './toolbar.component';
import { MatIconModule, MatToolbarModule } from '@angular/material';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('ToolbarComponent', () => {
	let fixture: ComponentFixture<ToolbarComponent>;
	let instance: ToolbarComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			declarations: [
				ToolbarComponent
			],
			imports: [
				MatIconModule,
				MatToolbarModule
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

		fixture = TestBed.createComponent(ToolbarComponent);
		instance = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should compile toolbar', () => {
		expect(fixture).toMatchSnapshot();
	});
});
