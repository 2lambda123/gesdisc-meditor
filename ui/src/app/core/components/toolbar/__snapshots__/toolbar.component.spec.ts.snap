// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarComponent should compile toolbar 1`] = `
<med-toolbar
  helpDocumentLocation={[Function String]}
>
  <mat-toolbar
    class="med-toolbar mat-toolbar mat-toolbar-single-row"
  >
    <img
      class="logo"
      routerlink="/"
      src="assets/logo.png"
    />
    <span
      class="spacer"
    />
    <med-login-status />
    <mat-menu
      fxhide.gt-sm=""
    >
      <button
        mat-menu-item=""
        routerlink="/"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          home
        </mat-icon>
         Home 
      </button>
      <a
        href="mailto:gsfc-uui-dev-disc@lists.nasa.gov"
        mat-menu-item=""
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          feedback
        </mat-icon>
         Feedback 
      </a>
      <a
        href="/assets/mEditorhelp.html"
        mat-menu-item=""
        target="_blank"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          help
        </mat-icon>
         Help 
      </a>
    </mat-menu>
    <button
      fxhide.gt-sm=""
      mat-icon-button=""
    >
      <mat-icon
        aria-hidden="true"
        class="mat-icon material-icons mat-icon-no-color"
        role="img"
      >
        more_vert
      </mat-icon>
    </button>
    <div
      fxhide.lt-md=""
    >
      <button
        class="tb-button"
        mat-button=""
        routerlink="/"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          home
        </mat-icon>
         Home 
      </button>
      <a
        class="tb-button"
        href="mailto:gsfc-uui-dev-disc@lists.nasa.gov"
        mat-button=""
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          feedback
        </mat-icon>
         Feedback 
      </a>
      <a
        class="tb-button"
        href="/assets/mEditorhelp.html"
        mat-button=""
        target="_blank"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons mat-icon-no-color"
          role="img"
        >
          help
        </mat-icon>
         Help 
      </a>
    </div>
  </mat-toolbar>
</med-toolbar>
`;
